GitCommands/Git/GitTreeParser.cs

```diff
@@ -0,0 +1,45 @@
+using System.Collections.Generic;
+using System.Linq;
+using GitUIPluginInterfaces;
+
+namespace GitCommands.Git
+{
+    public interface IGitTreeParser
+    {
+        IEnumerable<IGitItem> Parse(string tree);
+        GitItem ParseSingle(string rawItem);
+    }
+
+    public sealed class GitTreeParser : IGitTreeParser
+    {
```

Comment:
@jbialobr it would be good to have an integration test - e.g. run `git ls-tree hash -z` command and parse it... 
I don't know how, any help is appreciated